src/main/java/edu/kis/powp/jobs2d/drivers/decorator/DeviceUsageDecorator.java

```diff
@@ -10,8 +11,8 @@
 	private int posX = 0;
 	private int posY = 0;
 
-	public DeviceUsageDecorator(Job2dDriver driver, DeviceUsageManager deviceUsageManager) {
-		this.driver = driver;
+	public DeviceUsageDecorator(DriverManager driverManager, DeviceUsageManager deviceUsageManager) {
+		this.driver = driverManager.getCurrentDriver();
```

Comment:
The driver is fixed and cannot be changed after DeviceUsageDecorator construction. This modification doesn't change much regarding my suggestion: "It would be best to redesign Device Usage so it wouldn't use internal LineDriverAdapter anymore.". Actually, when using the driver composite approach there should be no need for "DeviceUsageDriver" to use the decorator pattern anymore. The driver field can be removed and still device usage should work.